kai/model_provider.py

```diff
@@ -78,6 +78,7 @@ def __init__(
             "ibm/granite-20b-multilang-lab-rc",
             "ibm/granite-20b-multilingual",
             "ibm/granite-3b-code-plus-v1",
+            "ibm/granite-20b-code-javaenterprise",
```

Comment:
I wonder if we should be validating them so strictly. Feel like it would make more sense to have a list of models we know work well, and issue a warning if someone is using a different model. Not on this PR but just a thought

Reply:
Totally agree, doesn't feel good to be this strict when the values are frequently changing.

Reply:
We have the below issues tracking to fix this

- Issue #129 
- Issue #164 (probably related)